python/audio_dsp/design/parse_json.py

```diff
@@ -204,51 +197,44 @@ def stage_handle(model):
     return getattr(Stages, model.op_type)
 
 
+def make_edge_name(node_name: str, idx: int) -> Tuple[str, int]:
+    """Return the canonical edge name as a tuple, e.g., ('NodeName', 0)."""
+    return (node_name, idx)
+
+
 def make_pipeline(json_obj: DspJson) -> Pipeline:
     """Create a Python DSP pipeline from a Pydantic model of the JSON file
     describing a DSP graph.
     """
     # Insert Fork nodes where needed to handle shared edges
-    graph = insert_forks(json_obj.graph)
-    # Flatten all input edge indices and count total channels
-    flat_input_edges: list[int] = []
+    # graph = insert_forks(json_obj.graph)
+    graph = json_obj.graph
+    edge_map = {}  # edge_name (tuple) -> pipeline object
+
+    # Collect all input edge names and total channels
+    flat_input_edges = []
     total_channels = 0
     for inp in graph.inputs:
-        flat_input_edges.extend(inp.output)
+        flat_input_edges.extend([tuple(e) for e in inp.output])
         total_channels += len(inp.output)
```

Comment:
We need to make the tuple from the name and channels
```suggestion
        flat_input_edges.extend([(inp.name, i) for i in range(inp.channels)])
        total_channels += inp.channels
```